src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py

```diff
@@ -1909,9 +1885,29 @@ def _make_session(fuzzer_name, job_type):
   return FuzzingSession(fuzzer_name, job_type, test_timeout)
 
 
+def do_multiarmed_bandit_strategy_selection(uworker_env):
+  """Set multi-armed bandit strategy selection during preprocessing. Set
+  multi-armed bandit strategy selection distribution as an environment variable
+  so we can access it in launcher."""
+  # TODO: Remove environment variable once fuzzing engine refactor is
+  # complete.
+  if not environment.get_value('USE_BANDIT_STRATEGY_SELECTION'):
+    return
+  selection_method = utils.random_weighted_choice(SELECTION_METHOD_DISTRIBUTION,
+                                                  'probability')
+  environment.set_value('STRATEGY_SELECTION_METHOD',
+                        selection_method.method_name, uworker_env)
+  distribution = get_strategy_distribution_from_ndb()
+  if not distribution:
+    return
+  environment.set_value('STRATEGY_SELECTION_DISTRIBUTION', distribution,
+                        uworker_env)
+
+
 def utask_preprocess(fuzzer_name, job_type, uworker_env):
-  session = _make_session(fuzzer_name, job_type)
-  session.preprocess()
+  do_multiarmed_bandit_strategy_selection(uworker_env)
+  environment.set_value('PROJECT_NAME', data_handler.get_project_name(job_type),
```

Comment:
Can we have a list of vars that we set in all jobs automatically? This seems generic enough that all tasks may want this.

Reply:
Are there more? I'd be happy to do this in another PR.